pkg/util/equality/podset.go

```diff
@@ -27,14 +27,17 @@ import (
 // TODO: Revisit this, maybe we should extend the check to everything that could potentially impact
 // the workload scheduling (priority, nodeSelectors(when suspended), tolerations and maybe more)
```

Comment:
Do we need more code to support changing node selectors, or can we just add the comparison now?

Reply:
It should work with adding the check only.

Reply:
However realigning the tests will take some effort.

Reply:
ok, it can be a follow up.